src/Pororoca.Desktop/ViewModels/WebSocketConnectionViewModel.cs

```diff
@@ -601,9 +615,19 @@ public async Task ConnectAsync()
             // causes the UI to freeze for a few seconds, especially when performing the first request to a server.
             // That is why we are invoking the code to run in a new thread, like below.
             await Task.Run(() => this.connector.ConnectAsync(resolvedClients.wsCli!, resolvedClients.httpCli!, resolvedUri!, this.cancelConnectionAttemptTokenSource.Token));
+            ResponseStatusCodeElapsedTimeTitle =
+                this.connector.ConnectionHttpStatusCode == HttpStatusCode.SwitchingProtocols
+                    ? FormatSuccessfulResponseTitle(this.connector.ElapsedConnectionTimeSpan,resolvedClients.wsCli!.HttpStatusCode)
+                    : FormatFailedResponseTitle(this.connector.ElapsedConnectionTimeSpan);
+            ConnectionResponseHeadersTableVm = new(this.connector.ConnectionHttpHeaders?.SelectMany(
```

Comment:
Instead of creating a new DataGridViewModel each time, you can clear its `Items` and add the new ones:

```cs
ConnectionResponseHeadersTableVm.Items.Clear();
var resHeaders = this.connector.ConnectionHttpHeaders?.SelectMany(
     kv => kv.Value.Select(val => new PororocaKeyValueParam(true, kv.Key, val))).ToList();
foreach (var header in resHeaders)
{
    ConnectionResponseHeadersTableVm.Items.Add(header);
}

```